dashbuilder/dashbuilder-client/dashbuilder-renderers/dashbuilder-renderer-c3/src/main/java/org/dashbuilder/renderer/c3/client/charts/pie/C3PieChartDisplayer.java

```diff
@@ -110,4 +119,8 @@ public DisplayerConstraints createDisplayerConstraints() {
                 .supportsAttribute(DisplayerAttributeDef.SUBTYPE);
     }
 
+    private C3Donut createDonut() {
+        String donutHoleTitle = displayerSettings.getDonutHoleTitle();
```

Comment:
@yesamer 
Hi. This is a sort of jump of the faith - to be sure that `displayerSettings` is not null (I do not see any direct reference in this class - maybe I'm wrong) and that its "donutHoleTitle" is not null too.
Beside that - if someone want/need to customize the `C3Donut` instantiation, he/she has to override the `buildConfiguration` method to invoke a different "createDonut" (being private, `createDonut` can't be overrided)
Just general observation, though.

Reply:
@gitgabrio In my opinion, the case where `displayerSettings` is null should be considered as an error, I consider it as a mandatory information to create a chart. That parameter is used to defines the properties of a chart (eg, renderer, chart type etc). All the logic flow makes this assumption, so I think it's correct in this contexts.  
Regarding the second point, I changed the modifier as you proposed.

Reply:
@yesamer 
Yes, I'm sure that `displayerSettings` is mandatory; at the same time, in this piece of code, it is actually potentially null (not this method, nor the containing class, has any control on it) - so I think a check - with a specific exception thrown, would be nice; if not in this specific method, at least in the class. Anyway, there is no common view about that.

Reply:
@gitgabrio A null check for that field is present in the super class which contains it: 'AbstractDisplayer.draw()' : 237. And this method it's called before the chart creation and eventually before the method with highlight. 
Anyway, this not completely avoid a NPE because, as you can see, 'displayerSettings` fields are used before that `draw()` (eg `initDisplayer()` method) but this is another story....